doc/memos/rdm0.md

```diff
@@ -0,0 +1,152 @@
+
+RIOT Developer Memos (RDM) are written in English, using the Markdown (.md) 
+format, with less than 80 characters per line. Each memo has a unique 
+identification number indicated in its header, 
+e.g. this memo is RDM0.
+
+The structure of a memo MUST contain:
+
+1. a preamble as defined below,
+2. the content of the memo structured as needed,
+3. a postamble as defined below.
+
+The preamble of the memo should be structured as such:
+
+1. a header formatted as the header of this memo,
+2. a short abstract,
+3. a status for the memo stating its license and the category of the memo (e.g. 
+that it is a consensus document),
+4. optionally, a terminology section which recalls/disambiguates definition of 
+terms that are key to the memo.
+
+The core content of the memo is structured as deemed necessary. However, during 
+the review process, care must be taken to follow the below principles as much
+as possible:
+
+1. one memo = one problem/aspect,
+2. start by crisply explaining and motivating the one problem/aspect the memo 
+targets,
+3. memos are expected to be of normative and durable nature, hence: keep a memo 
+as concise as possible without impairing clarity, and factor out content that 
+would be likely to be updated too often.
+
+The postamble of the memo should be structured as such:
+
+1. (optional) acknowledgements section thanking contributors other than the 
+authors,
+2. a section with the main references that relate to the content, with a strong
+preference for permanent, self-contained references,
+3. the list of revisions so far and corresponding changelog,
+4. contact information to reach the author(s).
+
+
+# 3. Memo Publishing and Maintenance Process
+
+Each new memo MUST be proposed as a pull-request (PR),
+with temporary number RDM 99999, 
+and discussed the same way RIOT code PRs are processed on GitHub.
+
+Proposing an RDM is non-compulsory. There is no restriction on when, in the 
+development cycle, an RDM can be PRed. However, RDMs on design decisions and 
+architecture is typically expected to describe something rather stable (what is 
+already merged in the main branch of RIOT, or about to be merged).
+
+Once proposed, a memo MUST have received an ACK by at least 3 maintainers other 
+than the author(s) of the memo, before its publication. 
+
+At this stage, the RDM is assigned a permanent RDM number (the next available),
+and merged in the master branch of RIOT. Once merged into the master branch
+of RIOT, a memo is considered published.
+
+Small changes/clarifications may be subsequently PRed to improve a memo, using 
+the same RDM number. In this case, a (monotically increasing) revision number 
+is issued and a changelog is added -- see Revision section in postamble below.
+
+Bigger changes should result in a new memo with another 
+RDM number, deprecating the old memo. In this case, the deprecated memo must 
+be modified to append a line in its header indicating "DEPRECATED BY RDM X" 
```

Comment:
Or we can have a "Status:" field similar to Python's PEPs.

Reply:
instead, the intent here was to mimic RFC style (see for example [rfc6918](https://tools.ietf.org/html/rfc6918) and
[rfc1788](https://tools.ietf.org/html/rfc1788) )

Reply:
I think Deprecated and non-Deprecated is enough. Python has a number of Statuses and I think this would be overkill - it'd add work and process, and thereby make it more difficult for us to support them

Reply:
Maybe also add a link then: "`[DEPRECATED BY RDM X](./rdmXXXX.md)`"?

Reply:
I was also thinking of Rust's RFC or Scheme SRFI. These go through a draft stage and then can become final or withdrawn.
My point is: how to have a solid proposal that is being worked on not stay in a PR? (without forcing it to be final)